r/src/expression.r

```diff
@@ -0,0 +1,45 @@
+#
+# runExpression  
+# returns the expression values for each gene in the input.
+# 
+# req$body has
+# genes: list of gene common names to search for in the annotation.  
+#
+runExpression <- function(req) {
+    #Get the annotation matrix with the geneid to name translation, and the subset with the correct names.
+    df <- data@misc$gene_annotations
+    genesSubset <- subset(df,df$name %in% req$body$genes)
+    #Get the expression values for those genes in the corresponding matrix.
+    geneExpression <- as.matrix(data@assays$RNA@data[genesSubset$input,])
+    #
+    #There might be a better way to do this, but when the count is 1 the previous
+    #search returns a transposed matrix without the gene id corresponding to that only column
+    #so I had to fix that.
+    #
+    if (ncol(geneExpression)==1){
+        geneExpression<-t(geneExpression)
+        for(gene in genesSubset$input){
+            if (gene %in% rownames(data@assays$RNA@data)){
```

Comment:
Normally, normalize data would be saved in data@assays$RNA, but in the case of integrated data, the slot keyword is going to be different. Hence, I suggest that we retrieve the counts directly from data@assays[[data@active.assay]]@data, where data@active.assay contains the slot name of the final matrix after normalization and integration.